plutus-ir/src/Language/PlutusIR/Compiler/Let.hs

```diff
@@ -43,7 +43,7 @@ compileLets kind t = getEnclosing >>= \p ->
 compileLet :: Compiling m e a => LetKind -> PIRTerm a -> DefT SharedName (Provenance a) m (PIRTerm a)
 compileLet kind = \case
     Let p r bs body -> withEnclosing (const $ LetBinding r p) $ case r of
-            NonRec -> lift $ foldM (compileNonRecBinding kind) body bs
+            NonRec -> lift $ foldM (compileNonRecBinding kind) body (reverse bs) -- this foldM is foldr 
```

Comment:
I'd say a bit more in this comment, something like "fold for lists is a foldl, but we want a foldr because we're compiling from the inside out, because we need things to be arranged like that as you can see from this example: ... ". Actually, sounds like this needs a note ;)